metaci/plan/migrations/0022_migrate_to_groups.py

```diff
@@ -64,23 +63,46 @@ def create_groups_and_migrate_users(apps, schema_editor):
 
 def grant_anonymous_perms(apps, schema_editor):
     PlanRepository = apps.get_model('plan.PlanRepository')
+    Group = apps.get_model('auth.Group')
     User = apps.get_model('users.User')
+    Permission = apps.get_model('auth.Permission')
+    ContentType = apps.get_model('contenttypes', 'ContentType')
+    GroupObjectPermission = apps.get_model('guardian.GroupObjectPermission')
+
+    public_group = Group.objects.create(name='Public')
+    view_builds = Permission.objects.get(codename='view_builds')
+
     try:
         anon = get_anonymous_user()
-    except Exception as e:
-        if not e.__class__.__name__ == 'DoesNotExist':
-            raise
+    except ObjectDoesNotExist:
         anon = User(
             username=guardian_settings.ANONYMOUS_USER_NAME,
             password=make_password(None),
         )
         anon.save()
+
+    if not anon.groups.filter(name='Public').exists():
+        anon.groups.add(public_group)
+  
+    content_type = ContentType.objects.get(app_label='guardian',model='groupobjectpermission') 
     for pr in PlanRepository.objects.all().iterator():
         if pr.plan.public is False:
             continue
         if pr.repo.public is False:
             continue
-        assign_perm('plan.view_builds', anon, pr)
+
+        obj_perm = GroupObjectPermission(
+            permission=view_builds,
+            group = public_group,
+            content_type = content_type,
```

Comment:
If this is a GenericForeignKey (I think it is?) this should be the ContentType for the object (in this case PlanRepository)

Reply:
...and it looks like its the ContentType for GroupObjectPermission itself, meaning the GFK will have confusion (at best)